### DISCUSSION TITLE
我们需要一个 FAQ！

### DISCUSSION THREAD

**Discussion by stdrc** [nonebot/discussions#10] [通用]
NoneBot 群里经常有新用户问相同或类似的问题，也许我们确实需要尽快有一个 FAQ，欢迎大家整理，然后发在 discussions～

Update: OK 我们现在有 FAQ 了，#13

**Comment by bytemain** [comment]
nonebot2 相关：
更新后无法使用 adapter
![image](https://user-images.githubusercontent.com/13938334/118070564-be2f2280-b3d8-11eb-80ae-bae3bdaaa0c3.png)

报错：no module named cqhttp

解决方案：
```
pip install nonebot-adapter-cqhttp
```

在 2.0 的某一次更新中，把 adapter 拆了出来单独发包，导致了这个问题。

**Reply by stdrc** [reply]
直接开一个新的帖子吧，然后 pin

**Comment by bytemain** [comment]
Nonebot 文档使用的主题是什么？
Vuepress

**Comment by StarHeartHunt** [comment]
nonebot2启动后无法连接上
检查host与port配置，检查端口占用，检查adapter是否注册
推荐先了解基本的C/S知识

**Comment by StarHeartHunt** [comment]
Linux怎么把bot挂到后台
用docker或者screen

**Comment by StarHeartHunt** [comment]
安装Python的注意事项
若为CentOS，编译后不建议覆盖python2的软链接，应注册一个不重名的软链接到编译出的bin文件夹，并添加bin目录到path以确保nb cli正常运行
若为Ubuntu，请使用apt进行安装Python3.8+以上的版本，确保nonebot2正常运行

**Comment by StarHeartHunt** [comment]
nonebot2启动后出现TypeError/Type xxx报错
请使用Python3.8+

**Comment by StarHeartHunt** [comment]
nonebot启动后插件不加载/无反应/配置未加载
请检查当前运行路径（`os.getcwd()`）是否为`bot.py`同级路径
检查ide的工作目录是否为bot.py同级目录

**Comment by StarHeartHunt** [comment]
Nonebot2写入日志就重载
`.env`中将`environment`设置为`prod`

**Comment by StarHeartHunt** [comment]
nonebot2如何开启冷重载
推荐使用`nb-cli`创建项目，初始配置默认开启冷重载

冷重载需要:
* `nonebot.run`传入app实例
* `.env`配置中设置`debug`为`true`
>

**Comment by StarHeartHunt** [comment]
apscheduler怎么用
<s>请立即阅读我们的文档</s>待完善

定时任务怎么拿bot实例（nonebot2）
`nonebot.get_bots()`

**Comment by StarHeartHunt** [comment]
nb-cli运行出错
* 出现`httpx`字样时，多为无法获取nonebot2的云端资源(插件/bot/adapter列表)或是代理服务器有问题，推荐切换或关闭代理后重试
* 请确保系统环境变量`path`中有`pip`的脚本目录

**Comment by StarHeartHunt** [comment]
nonebot初始化时出现ModuleNotFoundError: no module named xxx
请先同时卸载nonebot1和nonebot2：
```bash
pip uninstall nonebot
pip uninstall nonebot2
```
然后重新安装所需要的版本

**Comment by StarHeartHunt** [comment]
nonebot2升级adapter/本体后运行出现报错
尝试同时升级adapter与本体

**Comment by j1g5awi** [comment]
**我的协议端是 go-cqhttp，该选择 cqhttp adapter 还是 gocq adapter？**
    
gocq adapter 是非官方维护的适配器，对 go-cqhttp 的拓展 API 进行了适配，使得在 IDE 中可以获得更好的代码补全。
    
但如果你想获得**更稳定**的支持，请使用官方维护的 CQHTTP 适配器。

**Comment by stdrc** [comment]
大家直接在 https://github.com/nonebot/discussions/discussions/13 这个底下评论吧
